### PR TITLE
Print the Keys currently in use by IPsec

### LINKED ISSUE #28174: New cilium encrypt subcommands for managing keys
In our IPSec documentation, we tell users that they should rotate their keys, and provide a [handy but ugly](https://docs.cilium.io/en/stable/security/network/encryption-ipsec/#ipsec-key-rotation) shell script to do that.

It would be better if we had a command like: `cilium encrypt rotate-keys` that took care of this for the user.

Additionally, although perhaps less important, it would also be good to have `cilium encrypt key-status` that would print the data of the keys in use rather than having to use commands like `kubectl get secret -n kube-system cilium-ipsec-keys -o go-template --template={{.data.keys}} | base64 -d`.

### PR DESCRIPTION
**Pull request by chaunceyjiang**
Fixes: #28174

```release-note
Print the Keys currently in use by IPsec
```

### CODE REVIEW

**Comment by chaunceyjiang** [comment]
image

**Review by michi-covalent — APPROVED** [review]
change in cilium-dbg/cmd/encrypt_status.go lgtm ✅

File: pkg/common/ipsec/utils.go [hunk 1]
```diff
@@ -16,17 +18,22 @@ const (
...
 
 func CountUniqueIPsecKeys(states []netlink.XfrmState) int {
-	keys := make(map[string]bool)
+	return len(UniqueIPsecKeys(states))
+}
+
+func UniqueIPsecKeys(states []netlink.XfrmState) (keys []string) {
+	uniqueKeys := make(map[string]bool)
 	for _, s := range states {
 		if s.Aead == nil {
```
Comment (gentoo-root) [inline pkg/common/ipsec/utils.go:27]: Let's wait for https://github.com/cilium/cilium/pull/29182 to come in, it touches the same code and will conflict.

**Review by pchaigno — COMMENTED** [review]
Thanks for the pull request!

Could you explain why you want to display this information?

**Comment by chaunceyjiang** [comment]
> Could you explain why you want to display this information?

Hi @pchaigno  As described in the issue. https://github.com/cilium/cilium/issues/28174#issue-1898570092


> Additionally, although perhaps less important, it would also be good to have cilium encrypt key-status that would print the data of the keys in use rather than having to use commands like kubectl get secret -n kube-system cilium-ipsec-keys -o go-template --template={{.data.keys}} | base64 -d.

**Comment by pchaigno** [comment]
> This PR mainly aims to print the applied IPsec keys. Before this, only the number of applied IPsec keys was displayed, without showing the contents. It's not intended to implement `key rotations`.

I understand that. I'm wondering why we would need to print those IPsec keys. Even when we have a good use case, we typically avoid printing sensitive information like that.

**Pull request closed without merging**